src/Components/Components/src/Routing/Router.cs

```diff
@@ -248,19 +261,15 @@ internal async Task RunOnNavigateWithRefreshAsync(string path, bool isNavigation
             var previousTask = _previousOnNavigateTask;
             var tcs = new TaskCompletionSource(TaskCreationOptions.RunContinuationsAsynchronously);
             _previousOnNavigateTask = tcs.Task;
-            try
-            {
-                // And pass an indicator for the previous task to the currently running one.
-                var shouldRefresh = await RunOnNavigateAsync(path, previousTask);
-                if (shouldRefresh)
-                {
-                    Refresh(isNavigationIntercepted);
-                }
-            }
-            finally
+
+            // And pass an indicator for the previous task to the currently running one.
+            var shouldRefresh = await RunOnNavigateAsync(path, previousTask);
+            tcs.SetResult();
+            if (shouldRefresh)
             {
-                tcs.SetResult();
+                Refresh(isNavigationIntercepted);
             }
+
```

Comment:
I was a bit confused by the removal of the `try`/`finally` here since it looks like if the `RunOnNavigateAsync` task fails, then the router would now get stuck permanently, because `tcs` would never get marked as completed, and all subsequent navigation tasks will queue up behind it.

However on close inspection it looks like we're saying `RunOnNavigateAsync` won't fail, because the only place where we call user code from it is wrapped in a `try`/`catch`. If that's the reasoning then totally fine! Please let me know if I'm misreading it.